src/mgr/ActivePyModules.cc

```diff
@@ -1074,11 +1074,25 @@ void ActivePyModules::cluster_log(const std::string &channel, clog_type prio,
 {
   std::lock_guard l(lock);
 
-  if (channel == "audit") {
-    audit_clog->do_log(prio, message);
-  } else {
-    clog->do_log(prio, message);
-  }
+  auto cl = monc.get_log_client()->create_channel(channel);
+  map<string,string> log_to_monitors;
+  map<string,string> log_to_syslog;
+  map<string,string> log_channel;
+  map<string,string> log_prio;
+  map<string,string> log_to_graylog;
+  map<string,string> log_to_graylog_host;
+  map<string,string> log_to_graylog_port;
+  uuid_d fsid;
+  string host;
+  if (parse_log_client_options(g_ceph_context, log_to_monitors, log_to_syslog,
+			       log_channel, log_prio, log_to_graylog,
+			       log_to_graylog_host, log_to_graylog_port,
+			       fsid, host) == 0)
+    cl->update_config(log_to_monitors, log_to_syslog,
+		      log_channel, log_prio, log_to_graylog,
+		      log_to_graylog_host, log_to_graylog_port,
+		      fsid, host);
```

Comment:
```
<joao> sage, I think one would usually just init the channel once and reuse it,
thus diluting the init/config update step. In this case, given we're dealing with
arbitrary messages being logged to arbitrary channels, it might be a good idea
to include the whole init step in `LogClient::create_channel()`, init it once if the
channel does not exist, or return an existing channel. Might be better to have
the LogClient observing config changes though
<joao> as is, even if the steps themselves are not incredibly expensive, it's just
incredibly weird to see this being updated on every single logged message
```